kratos/includes/model_part.h

```diff
@@ -768,40 +768,35 @@ class KRATOS_API(KRATOS_CORE) ModelPart : public DataValueContainer, public Flag
     ///@name Properties
     ///@{
 
-    SizeType NumberOfProperties(IndexType ThisIndex = 0) const
-    {
-        return GetMesh(ThisIndex).NumberOfProperties();
-    }
+    /**
+     * @brief Returns the number of properties of the mesh
+     * @param ThisIndex The index identifying the mesh
+     * @return The number of properties of the mesh
+     */
+    SizeType NumberOfProperties(IndexType ThisIndex = 0) const;
 
-    /** Inserts a properties in the current mesh.
+    /**
+     * @brief Inserts a properties in the current mesh.
+     * @param pNewProperties The new property pointer to be added
+     * @param ThisIndex The index identifying the mesh
      */
     void AddProperties(PropertiesType::Pointer pNewProperties, IndexType ThisIndex = 0);
 
-    /** Returns if the Properties corresponding to it's identifier exists */
-    bool HasProperties(IndexType PropertiesId, IndexType MeshIndex = 0) const
-    {
-        auto pprop_it = GetMesh(MeshIndex).Properties().find(PropertiesId);
-        if(pprop_it != GetMesh(MeshIndex).Properties().end()) { // Property does exist
-            return true;
-        }
-
-        return false;
-    }
+    /**
+     * @brief Returns if the Properties corresponding to it's identifier exists
+     * @param PropertiesId The id identifying the property
+     * @param ThisIndex The index identifying the mesh
+     * @return True if the properties exist, false otherwise
+     */
+    bool HasProperties(IndexType PropertiesId, IndexType MeshIndex = 0) const;
 
-    /** Returns if the Properties corresponding to it's identifier exists in any of the model parts */
-    bool RecursivelyHasProperties(IndexType PropertiesId, IndexType MeshIndex = 0) const
-    {
-        auto pprop_it = GetMesh(MeshIndex).Properties().find(PropertiesId);
-        if(pprop_it != GetMesh(MeshIndex).Properties().end()) { // Property does exist
-            return true;
-        } else {
-            if(IsSubModelPart()) {
-                return mpParentModelPart->RecursivelyHasProperties(PropertiesId, MeshIndex);
-            } else {
-                return false;
-            }
-        }
-    }
+    /**
+     * @brief Returns if the Properties corresponding to it's identifier exists in any of the model parts
+     * @param PropertiesId The id identifying the property
+     * @param ThisIndex The index identifying the mesh
+     * @return True if the properties exist, false otherwise
+     */
+    bool RecursivelyHasProperties(IndexType PropertiesId, IndexType MeshIndex = 0) const;
```

Comment:
Why is this function needed? the root modelpart must have it...

Reply:
This method was already here. I just move it to the cpp (check it out)